showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/services/ItemService.java

```diff
@@ -20,14 +19,32 @@
         return itemDao.findAll();
     }
 
-    public Item findByItemID(Integer itemID) {
-        Optional<Item> item =itemDao.findById(itemID);
-        try{
-            item.get();
-        }catch(NoSuchElementException ex){
-            System.out.println("No user with that ID");
+    public Item addItem(Item item) {
+        return itemDao.saveAndFlush(item);
+    }
+
+    public void deleteItem(Integer id) {
+        itemDao.deleteById(id);
+    }
+
+    public Item getItem(Integer id) {
+
+        return itemDao.findById(id).get();
```

Comment:
This will throw a runtime exception if there is not an item with matching id. I would change it to something like this:
`
public Item findByItemID(Integer itemID) throws NoSuchItemException {
        Optional<Item> item =itemDao.findById(itemID);
        if(item.isPresent()) return item.get();
        else throw new NoSuchItemException("No item with id: " + itemID);
    }
`
Which is what I did in on my ErrorHandler branch, so we could just merge this and then merge my changes in and it should be fine.